src/analysis/kpi.ts

```diff
@@ -213,33 +207,199 @@ const getFirstDayOfMonth = (series: ISeries): Date => {
   );
 };
 
+export const filterDataframeByUrlWithPrefix = (
+  dataset: IDataFrame,
+  prefixUrl: string
+): IDataFrame => {
+  return dataset.where(
+    (log) => log.url != undefined && log.url.startsWith(prefixUrl)
+  );
+};
+
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export const computeCompletionRateOfUrlTool = (
-  logsOutilDataset: IDataFrame,
+  logs: IDataFrame,
+  startDate: Date,
   reportId: string = new Date().getTime().toString()
 ): KpiReport[] => {
-  const firstDay = getFirstDayOfMonth(
-    logsOutilDataset.getSeries("lastActionDateTime")
+  // Get logs on tools
+  const logsOutil = filterDataframeByUrlWithPrefix(
+    logs,
+    "https://code.travail.gouv.fr/outils"
   );
 
+  // Compute completion rate of convention collective tool first
   const conventionCollectiveCompletionRate =
-    getConventionCollectiveCompletionRate(logsOutilDataset, firstDay, reportId);
+    getConventionCollectiveCompletionRate(logsOutil, startDate, reportId);
 
-  const logsOutilDatasetFiltered = logsOutilDataset.where(
+  const logsOutilFiltered = logsOutil.where(
     (row) => row.outilAction == "view_step"
   );
 
-  const visitsByUrlAndEvent = getNumberOfVisitsByOutilAndEvent(
-    logsOutilDatasetFiltered
-  );
+  const visitsByUrlAndEvent =
+    getNumberOfVisitsByOutilAndEvent(logsOutilFiltered);
 
   const listOfKpisCompletionRate = getListOfKpiCompletionRate(
     visitsByUrlAndEvent,
-    firstDay,
+    startDate,
     reportId
   );
 
   listOfKpisCompletionRate.push(conventionCollectiveCompletionRate);
 
   return listOfKpisCompletionRate;
 };
+
+export const filterDataframeByContrib = (dataset: IDataFrame): IDataFrame => {
+  return filterDataframeByUrlWithPrefix(
+    dataset,
+    "https://code.travail.gouv.fr/contribution/"
```

Comment:
là aussi, mais ce que tu peux faire vu que c'est toujours la même url. ça serait faire ça : 

```suggestion
export const filterDataframeByContrib = (dataset: IDataFrame, url = "https://code.travail.gouv.fr/contribution/" ): IDataFrame => {
return filterDataframeByUrlWithPrefix(
    dataset,
    url
```